src/ComposerPlugin.php

```diff
@@ -227,7 +227,7 @@ private function detectGitDir(): void
     {
         $path = getcwd();
 
-        while (file_exists($path)) {
+        while (file_exists($path) && $path !== DIRECTORY_SEPARATOR) {
```

Comment:
What do you think about ` && !empty($path)` instead of `!== DIRECTORY_SEPARATOR` this would still allow `/.git` to work.

Reply:
This will still result in an endless '/' loop for me.

Reply:
Yes sure my fault `dirname('/')` returns `/`

Reply:
Where did I test that `dirname('/')` returned an empty string.... :weary:

Reply:
Then we should add something below line 236 like 
```php
// if we checked the root directory already, break to prevent endless loop
if ($path === dirname($path)) {
    break;
}
```

Reply:
I'm concerend though what weill happen on a windows-system....

Reply:
Not your fault @heiglandreas `git blame` said it was me 🙈

Reply:
> Then we should add something below line 236 like
> 
> ```
> // if we checked the root directory already, break to prevent endless loop
> if ($path === dirname($path)) {
>     break;
> }
> ```

If we add this below line 236 the loop will not check the top level directory.

Reply:
Nope. Because `$path` is then already '/' and has been checked. This condition will only be true on the second run through `/`

Reply:
@heiglandreas I'm pretty sure about this:

> CaptainHook Composer Plugin
> /Users/torhoehn/Downloads/captainhook
> /Users/torhoehn/Downloads
> /Users/torhoehn
> /Users
> 
> 
>   [RuntimeException]
>   Shiver me timbers! CaptainHook could not install yer git hooks! (git directory not found)
> 
>

Reply:
Right again, we would have to add it above line 236.... that's what I meant anyway *caugh* :P

Reply:
Updated my PR :)